index.html

```diff
@@ -553,9 +573,34 @@ <h1 class="title">The Ffau Editor</h1>
         }
     ]);
 
-    // add event listener
+    // add default event listener
     ffau.addEvent();
 
+    ffau.addEvent(() => {
+        if (ffau.editMode) {
+            try {
+                ffau.setXML(ffau.codeToXML(ffau.editor.getValue()));
+                ffau.ffauWorkspace.cleanUp();
+                ffau.ffauWorkspace.scrollCenter();
+            } catch (e) {
+                ffau.clearWorkspace();
+            }
+        }
+    }, 'ace');
+
+    ffau.workspaceDiv.onfocus = () => ffau.editMode = 0;
```

Comment:
Shouldn't this be a variable like `reverseCodeEnabled` with a boolean value?

Reply:
IMO that would be more ambiguous, as it would be unclear that `reverseCodeEnabled` also means that normal coding is disabled. To get around that, you could use another variable `blockGenerationEnabled`, with max. 1 being true at any point, but that also has more issues like what if they are both false. It's just not stable.